### PR TITLE
validate triggers also in the webhook

### PR DESCRIPTION
**Pull request by zroubalik**
Also moving `ScaleTriggers` to a separate file, because it is used both by ScaledObject and ScaledJob types.

### Checklist

- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes https://github.com/kedacore/keda/issues/4664

### CODE REVIEW

File: apis/keda/v1alpha1/scaledobject_webhook.go [hunk 1]
```diff
@@ -92,23 +92,39 @@ func isRemovingFinalizer(so *ScaledObject, old runtime.Object) bool {
...
+		verifyTriggers,
+		verifyScaledObjects,
+		verifyHpas,
 	}
-	err = verifyScaledObjects(so, action)
+
+	err := verifyCPUMemoryScalers(so, action)
 	if err != nil {
 		return nil, err
 	}
```
Comment (JorTurFer) [inline apis/keda/v1alpha1/scaledobject_webhook.go:106]: Isn't this duplicated? I mean, it's here and also in verifyFunctions

Comment (zroubalik) [reply apis/keda/v1alpha1/scaledobject_webhook.go:106]: Good catch! It's a leftover from copy&paste

Comment (zroubalik) [reply apis/keda/v1alpha1/scaledobject_webhook.go]: Fixed

**Review by JorTurFer — APPROVED** [review]

**Comment by zroubalik** [comment]
> /run-e2e **Update:** You can check the progress [here](https://github.com/kedacore/keda/actions/runs/5304222669)

@JorTurFer do you know if there's anything broken in following test in general?
```
Execution of tests/scalers/azure/azure_pipelines/azure_pipelines_test.go, has failed after "three" attempts
2023-06-18T15:54:03.0800698Z make: *** [Makefile:***08: e***e-test] Error ***
```

**Pull request merged**